src/Parser/AfflictionWarlock/Modules/Features/UABuffTracker.js

```diff
@@ -0,0 +1,125 @@
+
+import {UNSTABLE_AFFLICTION_DEBUFF_IDS} from '../../Constants';
+
+class UABuffTracker extends Module {
+  static dependencies = {
+    enemies: Enemies,
+  };
+
+  totalTicks = 0;
+  unbuffedTicks = 0;
+  ticksBuffedByReap = 0;
+  ticksBuffedByDrain = 0;
+  ticksBuffedByHaunt = 0;
+  ticksBuffedByBoth = 0;
+  hasMG = false;
+  hasHaunt = false;
+
+  on_initialized() {
+    if(!this.owner.error) {
+      this.hasMG = this.owner.selectedCombatant.hasTalent(SPELLS.MALEFIC_GRASP_TALENT.id);
+      this.hasHaunt = this.owner.selectedCombatant.hasTalent(SPELLS.HAUNT_TALENT.id);
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if(UNSTABLE_AFFLICTION_DEBUFF_IDS.indexOf(spellId) === -1)
+      return;
+    const target = this.enemies.getEntity(event);
+    this.totalTicks++;
+    const buffedByReap = this.owner.selectedCombatant.hasBuff(SPELLS.DEADWIND_HARVESTER.id, event.timestamp);
+    const buffedByDrain = target.hasBuff(SPELLS.DRAIN_SOUL.id, event.timestamp);
+    const buffedByHaunt = target.hasBuff(SPELLS.HAUNT.id, event.timestamp);
+
+    if(this.hasMG) {
+      if(buffedByReap && buffedByDrain)
+        this.ticksBuffedByBoth++;
+      else if(buffedByReap)
+        this.ticksBuffedByReap++;
+      else if(buffedByDrain)
+        this.ticksBuffedByDrain++;
+      else
+        this.unbuffedTicks++;
+    }
+    else if(this.hasHaunt) {
+      if(buffedByReap && buffedByHaunt)
+        this.ticksBuffedByBoth++;
+      else if(buffedByReap)
+        this.ticksBuffedByReap++;
+      else if(buffedByHaunt)
+        this.ticksBuffedByHaunt++;
+      else
+        this.unbuffedTicks++;
+    }
+    else {
+      if(buffedByReap)
+        this.ticksBuffedByReap++;
+      else
+        this.unbuffedTicks++;
+    }
+  }
+
+  suggestions(when){
+    const buffPercentage = this.unbuffedTicks / this.totalTicks;
+    when(buffPercentage).isGreaterThan(0.15)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest('Your Unstable Afflictions could be buffed more. Unstable Affliction is your main source of damage so keeping it buffed as much as possible with Reap Souls, Drain Soul (if using the Malefic Grasp talent) or Haunt (if using the talent) is very important.')
+          .icon(SPELLS.UNSTABLE_AFFLICTION_CAST.icon)
+          .actual(`${formatPercentage(actual)}% unbuffed Unstable Affliction ticks.`)
+          .recommended(`<${formatPercentage(recommended)}% is recommended`)
+          .regular(recommended + 0.05).major(recommended + 0.1);
+      });
+  }
+
+  statistic() {
+    const buffPercentage = 1 - (this.unbuffedTicks / this.totalTicks);
+    return (
+      <StatisticBox
+        icon={<SpellIcon id={SPELLS.UNSTABLE_AFFLICTION_CAST.id} />}
+        value={`${formatPercentage(buffPercentage)} %`}
+        label={(<dfn data-tip={`Your Unstable Afflictions ticked ${this.totalTicks} times in total. Out of that amount:
```

Comment:
I agree, I took inspiration from I think MW Monk module, it was exactly what I was looking for and I didn't know how to write it myself. I'll have a look at that.